File: node.js/src/slimmed/application/books/booksQueryService.ts
import { BookDetails } from '../../persistence/books/dtos';
import { FindDetailsByIdQuery } from './queries';
import { IBooksQueryRepository } from '../../persistence/books/repositories';

export interface IBooksQueryService {
  findDetailsById(query: FindDetailsByIdQuery): Promise<BookDetails | null>;
}

export class BooksQueryService implements IBooksQueryService {
  constructor(private repository: IBooksQueryRepository) {}

  findDetailsById(query: FindDetailsByIdQuery): Promise<BookDetails | null> {
    return this.repository.findDetailsById(query.data.bookId);
  }
}

File: csharp/Slimmed/PublishingHouse.Application/Books/BooksQueryService.cs
using PublishingHouse.Books.Entities;
using PublishingHouse.Persistence.Books.DTOs;
using PublishingHouse.Persistence.Books.Repositories;

namespace PublishingHouse.Application.Books;

public class BooksQueryService: IBookQueryService
{
    public Task<BookDetails?> FindDetailsById(BookId bookId, CancellationToken ct) =>
        repository.FindDetailsById(bookId, ct);

    public BooksQueryService(IBooksQueryRepository repository) =>
        this.repository = repository;

    private readonly IBooksQueryRepository repository;
}
